modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/ObjectEntryHelper.java

```diff
@@ -110,12 +117,145 @@ public ObjectEntry getObjectEntry(
+
+				entity.put(
+					property.getName(),
+					objectEntryProperties.get(objectField.getName()));
+			}
+
+			entities.add(entity);
+		}
+
+		return entities;
+	}
+
+	private Map<String, Object> _getAllObjectEntryProperties(
+		ObjectEntry objectEntry) {
+
+		return HashMapBuilder.<String, Object>putAll(
+			objectEntry.getProperties()
+		).put(
+			"createDate", objectEntry.getDateCreated()
+		).put(
+			"externalReferenceCode", objectEntry.getExternalReferenceCode()
+		).put(
+			"modifiedDate", objectEntry.getDateModified()
+		).build();
+	}
+
+	private Map<APIApplication.Property, ObjectField>
+			_getAPIApplicationPropertyObjectFieldMap(
+				long companyId, APIApplication.Schema schema,
+				ObjectDefinition objectDefinition)
+		throws Exception {
+
+		Map<APIApplication.Property, ObjectField> propertyObjectFieldHashMap =
+			new HashMap<>();
+
+		ObjectDefinition propertyObjectDefinition =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					"L_API_PROPERTY", companyId);
+
+		for (APIApplication.Property property : schema.getProperties()) {
+			ObjectEntry objectEntry1 = _objectEntryManager.getObjectEntry(
+				companyId, _getDefaultDTOConverterContext(objectDefinition),
+				property.getExternalReferenceCode(), propertyObjectDefinition,
+				null);
+
+			Map<String, Object> stringObjectMap = _getAllObjectEntryProperties(
+				objectEntry1);
+
+			propertyObjectFieldHashMap.put(
+				property,
+				_objectFieldLocalService.getObjectField(
+					(String)stringObjectMap.get("objectFieldERC"),
+					objectDefinition.getObjectDefinitionId()));
+		}
+
+		return propertyObjectFieldHashMap;
+	}
+
+	private DTOConverterContext _getDefaultDTOConverterContext(
+			ObjectDefinition objectDefinition)
+		throws Exception {
+
+		return new DefaultDTOConverterContext(
+			false, null, null, null, null, LocaleUtil.getSiteDefault(), null,
+			_userLocalService.getUser(objectDefinition.getUserId()));
+	}
+
+	private ObjectDefinition _getSchemaMainObjectDefinition(
+			long companyId, APIApplication.Schema schema)
+		throws Exception {
+
+		String objectEntryExternalReferenceCode =
+			schema.getExternalReferenceCode();
+
+		ObjectDefinition apiSchemaObjectDefinition =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					"L_API_SCHEMA", companyId);
+
+		Map<String, Object> objectEntryProperties =
+			_getAllObjectEntryProperties(
+				_objectEntryManager.getObjectEntry(
+					companyId,
+					new DefaultDTOConverterContext(
+						false, null, null, null, null,
+						LocaleUtil.getSiteDefault(), null,
+						_userLocalService.getUser(
+							apiSchemaObjectDefinition.getUserId())),
+					objectEntryExternalReferenceCode, apiSchemaObjectDefinition,
+					null));
+
+		return _objectDefinitionLocalService.
+			getObjectDefinitionByExternalReferenceCode(
+				(String)objectEntryProperties.get("mainObjectDefinitionERC"),
```

Comment:
Maybe it would be easier if we include this parameter as one of the Schema interface when parsing the ApiApplication:

```
public interface APIApplication {

        [...]

	public interface Schema {

		public String getDescription();

		public String getExternalReferenceCode();

		public String getName();

		public List<Property> getProperties();

	}

}
```

Reply:
Done